src/battle_ai_util.c

```diff
@@ -439,9 +439,9 @@ bool32 IsDamageMoveUnusable(u32 move, u32 battlerAtk, u32 battlerDef)
             return TRUE;
         break;
     case EFFECT_HIT_SET_REMOVE_TERRAIN:
```

Comment:
Actually, this change also break AI for the "Genesis Supernova" effect of setting terrain.

Reply:
Added a check for the effect's arguments.